www_src/components/option-panel/option-panel.jsx

```diff
@@ -0,0 +1,82 @@
+var React = require('react/addons');
+
+// Internal -- just the container
+var OptionPanel = React.createClass({
+  render: function () {
+    return (<div className="option-panel">
+      {this.props.children}
+    </div>);
+  }
+});
+
+var Checkbox = React.createClass({
+  mixins: [React.addons.LinkedStateMixin],
+  getDefaultProps: function () {
+    return {
+      linkState: this.linkState,
+      truthy: true,
+      falsey: false
+    };
+  },
+  onChange: function (e) {
+    var val = e.target.checked ? this.props.truthy : this.props.falsey;
+    this.valueLink.requestChange(val);
+  },
+  isTruthy: function () {
+    return this.valueLink.value === this.props.truthy;
+  },
+  render: function () {
+    var linkState = this.props.linkState;
+    this.valueLink = linkState(this.props.id);
+    return (<label className={'label ' + (this.isTruthy() ? 'selected' : '')}>
+      <input checked={this.isTruthy()} onChange={this.onChange} type="checkbox" />
+      <img className="icon" src={this.props.icon}/>
+    </label>);
+  }
+});
+
+var CheckboxSet = React.createClass({
+  mixins: [React.addons.LinkedStateMixin],
+  getDefaultProps: function () {
+    linkState: this.linkState
+  },
+  render: function () {
+    return (<OptionPanel>
+      {this.props.options.map(props => <Checkbox {...props} linkState={this.props.linkState} />)}
+    </OptionPanel>);
+  }
+});
+
+var Radio = React.createClass({
+  mixins: [React.addons.LinkedStateMixin],
+  getDefaultProps: function () {
+    linkState: this.linkState
+  },
+  onChange: function (e) {
+    console.log(e.target.value);
+    this.valueLink.requestChange(e.target.value);
+  },
+  render: function () {
+    this.valueLink = this.props.linkState(this.props.id);
+    var currentVal = this.valueLink.value;
+    return (<OptionPanel>
+      {this.props.options.map(item => {
+        return (<label className={'label ' + (item.value === currentVal ? 'selected' : '')}>
```

Comment:
missing `key` attributes? (fairly necessary if we want to take advantage of React's virtual DOM updating)